src/Sil/Component/Invoice/Model/Invoice.php

```diff
@@ -0,0 +1,590 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Copyright (C) 2015-2018 Libre Informatique
+ *
+ * This file is licenced under the GNU LGPL v3.
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Sil\Component\Invoice\Model;
+
+use Doctrine\Common\Collections\Collection;
+use Doctrine\Common\Collections\ArrayCollection;
+use Blast\Component\Resource\Model\ResourceInterface;
+
+/**
+ * @author Romain Sanchez <romain.sanchez@libre-informatique.fr>
+ */
+class Invoice implements ResourceInterface
+{
+    /**
+     * code.
+     *
+     * @var string
+     */
+    protected $code;
+
+    /**
+     * delivery date.
+     *
+     * @var DateTime
+     */
+    protected $date;
+
+    /**
+     * currency.
+     *
+     * @var CurrencyInterface
+     */
+    protected $currency;
+
+    /**
+     * status.
+     *
+     * @var InvoiceStatus
+     */
+    protected $status;
+
+    /**
+     * tax rate.
+     *
+     * @var float
+     */
+    protected $taxRate;
+
+    /**
+     * seller information.
+     *
+     * @var string
+     */
+    protected $sellerInfo;
+
+    /**
+     * due date.
+     *
+     * @var DateTime
+     */
+    protected $dueDate;
+
+    /**
+     * taxes.
+     *
+     * @var taxes
+     */
+    protected $taxes;
+
+    /**
+     * subtotal.
+     *
+     * @var float
+     */
+    protected $subTotal;
+
+    /**
+     * grand total.
+     *
+     * @var float
+     */
+    protected $grandTotal;
+
+    /**
+     * paid amount.
+     *
+     * @var float
+     */
+    protected $paidAmount;
+
+    /**
+     * due amount.
+     *
+     * @var float
+     */
+    protected $dueAmount;
+
+    /**
+     * payment terms.
+     *
+     * @var string
+     */
+    protected $paymentTerms;
+
+    /**
+     * created at.
+     *
+     * @var DateTime
+     */
+    protected $createdAt;
+
+    /**
+     * customer.
+     *
+     * @var CustomerInfoInterface
+     */
+    protected $customer;
+
+    /**
+     * invoice items.
+     *
+     * @var array|InvoiceItemInterface[]
+     */
+    protected $items;
+
+    /**
+     * invoice adjustments.
+     *
+     * @var array|InvoiceAdjustmentInterface[]
```

Comment:
Should be typed as * @var Collection|InvoiceAdjustmentInterface[] instead of array| ...